hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1452,17 +1449,13 @@ class ShrinkPass:
     index = attr.ib()
     shrinker = attr.ib()
 
-    next_prefix = attr.ib(default=())
-    fixed_point_at = attr.ib(default=None)
+    last_prefix = attr.ib(default=())
     successes = attr.ib(default=0)
     calls = attr.ib(default=0)
     shrinks = attr.ib(default=0)
     deletions = attr.ib(default=0)
 
     def step(self):
-        if self.fixed_point_at is self.shrinker.shrink_target:
-            return False
-
```

Comment:
I assume the idea here is that a pass is only marked as being at fixpoint when its choice tree is exhausted, in which case the check below will fail anyway.

(And a successful shrink will implicitly clear all the choice trees anyway, so there's no need to explicitly monitor the current target.)

Reply:
Correct. This check could only ever have fired when the next check would immediately have returned False anyway.